spec/services/charts/home/categories_with_changes_date_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+describe Charts::Home::CategoriesWithChangesDate do
+  let(:user) { create(:user) }
+
+  let(:balance) { Faker::Number.decimal(4, 2).to_f }
+  let(:date) { Faker::Date.between(1.year.ago, Date.current) }
+  let(:balance_params) { {date: date, amount: balance, user_id: user.id, comment: nil} }
+
+  let(:category) { create(:category, user: user, categorizable: user) }
+  let(:amount) { Faker::Number.decimal(2, 2).to_f }
+
+  let(:categories_titles) { categories.pluck(:title) }
+
+  let(:changes) do
+    Charts::Home::CategoriesWithChanges.new(Transaction.category_transactions).call
+  end
+
+  let(:data_chenges) { described_class.new(changes).date }
+
+  before do
+    BalanceTransactions::Creator.new(balance_params).create
```

Comment:
@marderer why didn't you use factories for that? Update you factories with such logic if needed